From 3b04eaddf7919fec99b89f3d201eee3636bdd1b6 Mon Sep 17 00:00:00 2001
Message-ID: <3b04eaddf7919fec99b89f3d201eee3636bdd1b6.1708471786.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1708471786.git.sharlatanus@gmail.com>
References: <cover.1708471786.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 27 Jan 2024 16:47:30 +0000
Subject: [PATCH 34/46] gnu: python-regions: Speed up tests.

* gnu/packages/astronomy.scm (python-regions): Enable parallel
tests to speed them up.
[arguments] <#:test-flags>: Add it with "-n" option.
[native-inputs]: Add python-pytest-xdist.

Change-Id: Ib0c50fa7c357c70ea78d803428a39f6d6ec3c2e7
---
 gnu/packages/astronomy.scm | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 0a2599590a..7b8160a3ea 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -2451,6 +2451,8 @@ (define-public python-regions
     (build-system pyproject-build-system)
     (arguments
      (list
+      #:test-flags
+      #~(list "-n" "auto")
       #:phases
       #~(modify-phases %standard-phases
           ;; This file is opened in both install and check phases.
@@ -2475,6 +2477,7 @@ (define-public python-regions
            python-pytest-arraydiff
            python-pytest-astropy
            python-pytest-runner
+           python-pytest-xdist
            python-setuptools-scm))
     (home-page "https://github.com/astropy/regions")
     (synopsis "Package for region handling")
-- 
2.41.0

